tabrepo/utils/huggingfacehub_utils.py

```diff
@@ -0,0 +1,115 @@
+    Returns:
+        None
+    """
+    commit_message = f"Upload tabrepo new version"
+    root = Path(__file__).parent.parent.parent / f"data/results/{version}/"
+
+    for filename in ["baselines.parquet", "configs.parquet", "model_predictions"]:
+        assert (root / filename).exists(), f"Expected to found {filename} but could not be found in {root / filename}."
+    api = HfApi()
+    for filename in ["baselines.parquet", "configs.parquet"]:
+        path_in_repo = str(Path(version) / filename)
+        if api.file_exists(repo_id=repo_id, filename=path_in_repo, token=os.getenv("HF_TOKEN"), repo_type="dataset") and not override_existing_files:
+            print(f"Skipping {path_in_repo} which already exists in the repo.")
+            continue
+
+        api.upload_file(
+            path_or_fileobj=root / filename,
+            path_in_repo=path_in_repo,
+            repo_id=repo_id,
+            repo_type="dataset",
+            commit_message=commit_message,
+            token=os.getenv("HF_TOKEN"),
+        )
+    files = list(sorted(Path(root / "model_predictions").glob("*")))
+    for dataset_path in tqdm(files):
+        print(dataset_path)
+        try:
+            path_in_repo = str(Path(version) / "model_predictions" / dataset_path.name)
+            # ideally, we would just check if the folder exists but it is not possible AFAIK, we could alternatively
+            # upload per file but it would create a lot of different commits.
+            if api.file_exists(repo_id=repo_id, filename=str(Path(path_in_repo) / "0" / "metadata.json"), token=os.getenv("HF_TOKEN"),
+                               repo_type="dataset") and not override_existing_files:
+                print(f"Skipping {path_in_repo} which already exists in the repo.")
+                continue
+            api.upload_folder(
+                folder_path=dataset_path,
+                path_in_repo=path_in_repo,
+                repo_id=repo_id,
+                repo_type="dataset",
+                ignore_patterns="*DS_Store",
+                commit_message=f"Upload tabrepo new version {dataset_path.name}",
+                token=os.getenv("HF_TOKEN"),
+            )
+        except Exception as e:
+            print(str(e))
+
+
+def download_from_huggingface(
+        version: str = "2023_11_14",
+        force_download: bool = False,
+        local_files_only: bool = False,
+        datasets: list[str] | None = None,
+):
+    """
+    :param version: name of a tabrepo version such as `2023_11_14`
+    :param local_files_only: whether to use local files with no internet check on the Hub
+    :param force_download: forces files to be downloaded
+    :param datasets: list of datasets to download, if not specified all datasets will be downloaded
+    :return:
+    """
+    # https://huggingface.co/datasets/Tabrepo/tabrepo/tree/main/2023_11_14/model_predictions
+    api = HfApi()
+    local_dir = str(Path(__file__).parent.parent.parent / "data/results/")
```

Comment:
We should be able to specify a local_dir as an argument

Reply:
Makes sense, will update to make this flexible.